### DISCUSSION TITLE
Upload 使用自带的上传方法可以添加响应拦截吗？

### DISCUSSION THREAD

**Discussion by pangxzi** [ant-design/ant-design#41561] [Q&A]
有的数据需要提前处理一下

![1680169127752](https://user-images.githubusercontent.com/52907799/228796011-b62d1be1-bf1f-4620-accc-9bcd987bbd6d.jpg)

**Comment by zombieJ** [comment]
`beforeUpload`

**Reply by pangxzi** [reply]
> `beforeUpload`

这个不是上传文件之前的方法吗？我想在响应拦截的时候修改response的数据

**Reply by pangxzi** [reply]
> `beforeUpload`

需要使用 `transformResponse` 处理一下数据

**Reply by zombieJ** [reply]
`beforeUpload` 可以拦截自行处理，不用 Upload 原来的上传逻辑。自然返回的内容你也可以自己处理了哈

**Reply by pangxzi** [reply]
> `beforeUpload` 可以拦截自行处理，不用 Upload 原来的上传逻辑。自然返回的内容你也可以自己处理了哈

我需要在form里面使用Upload，使用`beforeUpload`之后好像`onchange`就不回返回文件

**Reply by zombieJ** [reply]
HOC 包一下 Upload 代理掉 value 和 onChange 事件
